hs_core/hydro_realtime_signal_processor.py

```diff
@@ -0,0 +1,96 @@
+from django.db import models
+from haystack.signals import RealtimeSignalProcessor
+from haystack.exceptions import NotHandled
+import logging
+import types
+from haystack.query import SearchQuerySet
+from haystack.utils import get_identifier
+
+logger = logging.getLogger(__name__)
+
+
+class HydroRealtimeSignalProcessor(RealtimeSignalProcessor):
+
+    """ 
+    Customized for the fact that all indexed resources are subclasses of BaseResource. 
+
+    Notes: 
+    1. RealtimeSignalProcessor already plumbs in all class updates. We might want to be more specific. 
+    2. The class sent to this is a subclass of BaseResource, or another class. 
+    3. Thus, we want to capture cases in which it is an appropriate instance, and respond. 
+    """
+
+    def hydro_resource_is_present(self, instance): 
```

Comment:
I don't believe doing a round-trip to the Solr server before doing an indexing operation is a worthwhile thing to do.  I would recommend instead basing the decision on whether to index / remove from index solely on the information available in on the model, and if it happens that the item isn't index, catch the error (if necessary, haystack may be already dealing with such things relatively quietly) and move on.  This is definitely a case of "better to ask forgiveness than permission".